src/eddymotion/cli/parser.py

```diff
@@ -0,0 +1,113 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2024 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Parser module."""
+
+from argparse import ArgumentDefaultsHelpFormatter, ArgumentParser, Namespace
+from pathlib import Path
+from typing import Optional
+
+
+def _build_parser() -> ArgumentParser:
+    """
+    Build parser object.
+
+    Returns
+    -------
+    ArgumentParser
+        The parser object configured for the command-line arguments.
+    """
+    parser = ArgumentParser(
+        description="A model-based algorithm for the realignment of dMRI data.",
+        formatter_class=ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "input_dir",
+        action="store",
+        type=Path,
+        help="Path to the directory containing the original DWI data.",
+    )
```

Comment:
I don't know how this is workable. I think the easiest would be to require a single HDF5 file at the input.

Then we can create another command line to help the user go from NIfTIs to a single HDF5 file.

Reply:
I clarified the format of the input in https://github.com/nipreps/eddymotion/pull/157/commits/46c9427f98f8b5feda11efa0e44c6adbac28c8a5.